src/pages/ExportTransactions/index.tsx

```diff
@@ -0,0 +1,293 @@
+import { InputNumber, Tabs } from 'antd'
+import DatePicker from 'antd/lib/date-picker'
+import * as cnLocale from 'antd/es/date-picker/locale/zh_CN'
+import * as enLocale from 'antd/es/date-picker/locale/en_US'
+import dayjs, { Dayjs } from 'dayjs'
+import { useState } from 'react'
+import classNames from 'classnames'
+import { useTranslation } from 'react-i18next'
+import { useSearchParams, useUpdateSearchParams } from '../../utils/hook'
+import Content from '../../components/Content'
+import styles from './styles.module.scss'
+import 'dayjs/locale/es-us'
+import 'dayjs/locale/zh-cn'
+import { ReactComponent as BlockIcon } from '../../assets/block_icon.svg'
+import { ReactComponent as ErrorIcon } from '../../assets/error_icon.svg'
+import { ReactComponent as SuccessIcon } from '../../assets/success_icon.svg'
+import { omit } from '../../utils/object'
+import { exportTransactions } from '../../service/http/fetcher'
+
+const ExportTransactions = () => {
+  const [t, { language }] = useTranslation()
+
+  const locale = language === 'zh' ? cnLocale.default : enLocale.default
+
+  const {
+    type: typeStr,
+    id,
+    tab = 'date',
+    'start-date': startDateStr,
+    'end-date': endDateStr,
+    'from-height': fromHeightStr,
+    'to-height': toHeightStr,
+  } = useSearchParams('type', 'id', 'tab', 'start-date', 'end-date', 'from-height', 'to-height')
+
+  function isTransactionCsvExportType(s?: string): s is State.TransactionCsvExportType {
+    return s ? ['address_transactions', 'blocks', 'udts', 'nft'].includes(s) : false
```

Comment:
I think maybe `return ['address_transactions', 'blocks', 'udts', 'nft'].includes(s)` is enough?

Reply:
It's a flaw. Maybe the origin author want to pass static type check, which restrains s must be non-nullable.
I have updated it to
```ts
return !!s && ['address_transactions', 'blocks', 'udts', 'nft'].includes(s)
```